[Justin Aw] iP

Justin-Aw46 opened the pull request:
# Papagu 

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Papagu frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/149).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Adding tasks (Todos, Deadlines, Events)
- [ ] Managing the tasks
- [ ] Checking completion of tasks
- [ ] Searching for specific tasks
- [ ] Checking your todo list

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:

```Java
public class Main {
    public static void main(String[] args) {
        new Papagu("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Justin-Aw46 committed:
Rename main class to Papagu

Justin-Aw46 committed:
Add greeting and exit

Justin-Aw46 committed:
Add echo functionality

Justin-Aw46 committed:
Add TaskList class

Justin-Aw46 committed:
Add addTask method

Justin-Aw46 committed:
Override toString method

Justin-Aw46 committed:
Add lines to be printed in addTask

Justin-Aw46 committed:
Add support for echo with TaskList

Justin-Aw46 committed:
Add Task class

Justin-Aw46 committed:
Incorporate Task class into TaskList class

Justin-Aw46 committed:
Add setters to mark if Task is completed in TaskList

Justin-Aw46 committed:
Add Mark as Done functionality

Justin-Aw46 committed:
Fix bug in indexing for TaskList

Justin-Aw46 committed:
Add ToDos class

Justin-Aw46 committed:
Add Deadlines class

Justin-Aw46 committed:
Add Events class

Justin-Aw46 committed:
Modify TaskList to take in Tasks

Justin-Aw46 committed:
Add functionality to add different types of tasks

Justin-Aw46 committed:
Edit text displayed when methods are called

Justin-Aw46 committed:
Fix bug with marking event and edited text that is displayed

Justin-Aw46 committed:
Change name of main file

Justin-Aw46 committed:
Add test cases to input

Justin-Aw46 committed:
Add expected values to test cases

Justin-Aw46 committed:
Create exception to handle any incorrect inputs

Justin-Aw46 committed:
Create exception to handle todo events with empty description

Justin-Aw46 committed:
Created exception to handle Deadline event with no description

Justin-Aw46 committed:
Create exception to handle Event with no description

Justin-Aw46 committed:
Add try catch block and exception throwing to handle incorrect inputs

Justin-Aw46 committed:
Add DeleteTask to TaskList

Justin-Aw46 committed:
Add Delete functionality to main class

Justin-Aw46 committed:
Add Tasks.txt

Justin-Aw46 committed:
Revert "Add Tasks.txt"

This reverts commit fb4b9552f11601e99aa6a1eb61365a33c3b4560b.

Justin-Aw46 committed:
Import java.io.file

Justin-Aw46 committed:
Handle file reading for todos

Justin-Aw46 committed:
Add getTask method

Justin-Aw46 committed:
Fix file reading

Justin-Aw46 committed:
Add toFile method for Deadline class

Justin-Aw46 committed:
Add toFile method for Events

Justin-Aw46 committed:
Add toFile method for todos class

Justin-Aw46 committed:
Add isDone and getDescription and toFile for task class

Justin-Aw46 committed:
Handle mark and unmark commands to write file

Justin-Aw46 committed:
Handle writing to file for todo command

Justin-Aw46 committed:
Handle file writing for deadlines command

Justin-Aw46 committed:
handle file writing for events class

Justin-Aw46 committed:
Test how file reading works

Justin-Aw46 committed:
Add the Tasks.txt to gitignore

Justin-Aw46 committed:
Edit Deadlines class to take in LocalTime and LocalDate input

Justin-Aw46 committed:
Edit events class to handle LocalDate and LocalTime inputs

Justin-Aw46 committed:
edit main class to handle local date and time inputs as well as storage

Justin-Aw46 committed:
Create Parser file

Justin-Aw46 committed:
Create Storage file

Justin-Aw46 committed:
Create Ui file

Justin-Aw46 committed:
Add method to parse text from tasks.txt file

Justin-Aw46 committed:
Add logic to Parse class

Justin-Aw46 committed:
changed Papagu to utilise other classes

Justin-Aw46 committed:
Add logic to storage class

Justin-Aw46 committed:
Add print functions to Ui Class

Justin-Aw46 committed:
Remove print functions from tasklist

Justin-Aw46 committed:
Change input.txt to match current inputs

Justin-Aw46 committed:
Move to new folder for package

Justin-Aw46 committed:
Add package statement

Justin-Aw46 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

* commit '81a9c537afe8492675eb253968939364d65db091':
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Justin-Aw46 committed:
Add Test file

Justin-Aw46 committed:
Edit build file

Justin-Aw46 committed:
Add tests for welcome message, list, and event commands

Justin-Aw46 committed:
Add some code to test for file path

Justin-Aw46 committed:
Edit files for jar creation

Justin-Aw46 committed:
Revert "Edit files for jar creation"

This reverts commit 0af27011144528a5399886de23ef525b05b92a7c.

Justin-Aw46 committed:
Add JavaDocs for all classes and methods

Justin-Aw46 committed:
Edit certain lines to fit coding standard

Justin-Aw46 committed:
Add method to check for match in given string with task description

Justin-Aw46 committed:
Chang return type of findTasks

Justin-Aw46 committed:
Add print function for find command

Justin-Aw46 committed:
Add method to find matching task in the tasklist

Justin-Aw46 committed:
Edit the javadoc for isMatching

Justin-Aw46 committed:
Edit parser to include find command

Justin-Aw46 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Edit certain lines to fit coding standard

Justin-Aw46 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Edit parser to include find command
  Edit the javadoc for isMatching
  Add method to find matching task in the tasklist
  Add print function for find command
  Chang return type of findTasks
  Add method to check for match in given string with task description

sheenkerr commented on build.gradle:
```diff
@@ -0,0 +1,42 @@
+}
+
+shadowJar {
+    archiveBaseName = "duke"
```
I think you need to change this for the A-Jar branch later on

sheenkerr commented on src/main/java/Papagu/Ui/Events.java:
```diff
@@ -0,0 +1,56 @@
+ */
+public class Events extends Task {
+    private LocalDate date;
+    private LocalTime start;
```
Maybe you could use more descriptive names like `startTime` & `endTime`

sheenkerr commented on src/main/java/Papagu/Ui/FileTest.java:
```diff
@@ -0,0 +1,15 @@
+package Papagu.Ui;
+
```
You may want to import JUnit to use Gradle to run your tests

sheenkerr commented on src/main/java/Papagu/Ui/Papagu.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public static void main(String[] args) {
+        // Print current working directory
```
Maybe remove the un-needed comments as it makes the code look a bit messy

sheenkerr commented on src/main/java/Papagu/Ui/Parser.java:
```diff
@@ -0,0 +1,233 @@
+     * @return
+     */
+    private static int monthConverter(String month) {
+        switch (month) {
```
I think this does not follow the coding standards, as switch and case should have the same level of indentation

sheenkerr reviewed (commented):
LGTM. Just some formatting to fix. 

Also I noticed that all your Java Doc's `@return` are blank so you may want to update those as well. 😃

Justin-Aw46 committed:
Edit the switchcases to follow coding standard

Justin-Aw46 committed:
Edit build.gradle to support checkstyle

Justin-Aw46 committed:
Add files for checkstyle

Justin-Aw46 committed:
update build.gradle to support GUI

Justin-Aw46 committed:
Update files to handle GUI

Justin-Aw46 committed:
Add FXML support for GUI

Justin-Aw46 committed:
remove unused variable

Justin-Aw46 committed:
add assertion for file to exit

Justin-Aw46 committed:
assert string input has correct format

Justin-Aw46 committed:
Add assertion for input parsing

This is to check that each input has the correct format of inputs,
making sure that when accessing the different parts of the input there
will not be any errors

Just in case there are some wrong inputs in the file

Justin-Aw46 committed:
Shorten function body for todo and deadline

Created a new function for each of the events todo and deadline
such that it would shorten the method body for parseFile

allows for easier readability with there being a specific function
for each event

Justin-Aw46 committed:
Shorten the function body for input parsing

Justin-Aw46 committed:
Merge pull request #2 from Justin-Aw46/branch-A-Assertions

Branch A assertions

Justin-Aw46 committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertion for input parsing
  assert string input has correct format
  add assertion for file to exit

Justin-Aw46 committed:
Fix bug - wrong variable name

Justin-Aw46 committed:
Add help message and functionality

Justin-Aw46 committed:
Add final changes for submission

Justin-Aw46 committed:
fix bug with september not parsing correctly

Justin-Aw46 committed:
gitignore the tasks.txt file

Justin-Aw46 committed:
Update the help command

Justin-Aw46 committed:
Add screenshot

Justin-Aw46 committed:
Edit readme

Justin-Aw46 committed:
Make slight edits to remove redundant ccomments and change UI output

Justin-Aw46 committed:
Add README

Justin-Aw46 committed:
Change some comments to match javadocs

Justin-Aw46 committed:
Make more edits for coding standard

Justin-Aw46 committed:
Fix checkstyle errors

Justin-Aw46 committed:
Update ui for text warping

Justin-Aw46 committed:
fix tests

Justin-Aw46 committed:
Add Ui picture
